server/qrops.go

```
@@ -0,0 +1,281 @@
+package server
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"image"
+	"net/url"
+	"strconv"
+
+	"github.com/yeqown/go-qrcode/v2"
+	"github.com/yeqown/go-qrcode/writer/standard"
+	"go.uber.org/zap"
+
+	"github.com/status-im/status-go/images"
+	"github.com/status-im/status-go/multiaccounts"
+)
+
+type WriterCloserByteBuffer struct {
+	*bytes.Buffer
+}
+
+func (wc WriterCloserByteBuffer) Close() error {
+	return nil
+}
+
+func NewWriterCloserByteBuffer() *WriterCloserByteBuffer {
+	return &WriterCloserByteBuffer{bytes.NewBuffer([]byte{})}
+}
+
+type QRConfig struct {
+	DecodedQRURL    string
+	WithLogo        bool
+	CorrectionLevel qrcode.EncodeOption
+	KeyUID          string
+	ImageName       string
+	Size            int
+	Params          url.Values
+	Logger          *zap.Logger
+}
+
+func NewQRConfig(params url.Values, logger *zap.Logger) *QRConfig {
+	config := &QRConfig{}
+	config.Params = params
+	config.Logger = logger
+	config.SetQrURL()
+	config.SetAllowProfileImage()
+	config.SetErrorCorrectionLevel()
+	config.SetSize()
+
+	if config.WithLogo {
+		config.SetKeyUID()
+		config.SetImageName()
+	}
+
+	return config
+}
+
+func (q *QRConfig) SetQrURL() {
+	qrURL, ok := q.Params["url"]
+	logger := q.Logger
+
+	if !ok || len(qrURL) == 0 {
+		logger.Error("[qrops-error] no qr url provided")
```

if it's not that serious, maybe using logger.Warn or something else would be more suitable?

---

Hmm, the `url` is a required parameter and hence this is a critical error.
Nothing will work if `url` is not provided or if an empty string is passed for this parameter from client side.

---

If this is a critical error you should probably return an error at this point to inform the consuming code.

---

fixed here : https://github.com/status-im/status-go/pull/3172/commits/b55243dbf691338c82ca6c09e00042f7c8decfe1